### ISSUE TITLE
should `string[]` elements be nilable?

### ISSUE THREAD

**Issue by lewis6991** [CppCXY/emmylua-analyzer-rust#248]
```lua
local a --- @type string[]
local b, c = unpack(a)
-- type string, string
-- should it actually be string?, string?
```

For context this is because of this code:

<img width="827" alt="Image" src="https://github.com/user-attachments/assets/cf9fc839-4672-49f7-b035-c73c1abd4544" />

If we decide the current behaviour should stay the same, then I'll need to add a `@diagnostic disable` on this assert. Alternatively `vim.split` could return `(string?)[]`

**Comment by lewis6991** [comment]
Another situation:

```lua
local stdout --- @type string[]

local line1 = assert(stdout[1]) -- causes unnecessary assert diagnostic
```

We don't know the length of `stdout`, so any index may be `nil`.

**Comment by lewis6991** [comment]
array indexes are now nilable
